robottelo/hosts.py

```diff
@@ -1416,11 +1415,16 @@ def register_to_cdn(self, pool_ids=None, enable_proxy=False):
             raise ContentHostError(
                 f'Error during registration, command output: {cmd_result.stdout}'
             )
-        cmd_result = self.subscription_manager_attach_pool(pool_ids)[0]
-        if cmd_result.status != 0:
-            raise ContentHostError(
-                f'Error during pool attachment, command output: {cmd_result.stdout}'
-            )
+        # Attach a pool only if the Org isn't SCA yet
+        sub_status = self.subscription_manager_status().stdout
+        if SM_OVERALL_STATUS['disabled'] not in sub_status:
+            if pool_ids is None:
+                pool_ids = [settings.subscription.rhn_poolid]
+            cmd_result = self.subscription_manager_attach_pool(pool_ids)[0]
```

Comment:
wait. only now i realized - we accept a list of pool ids but finally only use the first one? hasn't this been a bug in the method?

Reply:
Yeah, this is how it was [since beginning](https://github.com/SatelliteQE/robottelo/commit/f54efe9c62009473808b6eb1f473ccb9f176e466). I don't know the reason behind, we could ask @Gauravtalreja1 or we don't need to slam there `assert all` and that's it. But again, the pools are going away soon ... unreachable check.